xdsl/interpreters/eqsat_pdl.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import IO, Any, cast
+
+from xdsl.context import MLContext
+from xdsl.dialects import eqsat, pdl
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.interpreter import Interpreter, impl, register_impls
+from xdsl.interpreters.pdl import PDLMatcher, PDLRewriteFunctions
+from xdsl.ir import Attribute, Operation, OpResult, SSAValue, TypeAttribute
+from xdsl.irdl import IRDLOperation
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.rewriter import InsertPoint
+from xdsl.transforms.convert_onnx_to_linalg import get_root_op
+from xdsl.utils.exceptions import InterpretationError
+
+
+@dataclass
+class EqsatPDLMatcher(PDLMatcher):
+    def match_operand(
+        self, ssa_val: SSAValue, pdl_op: pdl.OperandOp, xdsl_val: SSAValue
+    ):
+        owner = xdsl_val.owner
+        assert isinstance(owner, eqsat.EClassOp)
+        assert (
+            len(owner.operands) == 1
+        ), "newly converted eqsat always has 1 element in eclass"
+        arg = owner.operands[0]
+        res = super().match_operand(ssa_val, pdl_op, arg)
+        # self.value_to_eclass[arg] = owner
+        return res
```

Comment:
could you please open a first PR that just adds this, and adds a corresponding pytest unit test for this in the style of the existing pdl matching tests?

Reply:
Do you mean splitting this PR into two (Matcher and Rewriter)?

Reply:
Created an PR here: https://github.com/xdslproject/xdsl/pull/3574

Reply:
@cowardsa 
I think this PR only targets the example of swapping inputs for (a+b) to help review. Future PRs can target (a+b)+c progressively.